source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -10,16 +11,20 @@ namespace HttpFilters {
 namespace OnDemand {
 
 Http::FilterHeadersStatus OnDemandRouteUpdate::decodeHeaders(Http::RequestHeaderMap&, bool) {
-  if (callbacks_->route() != nullptr ||
-      !(callbacks_->routeConfig().has_value() && callbacks_->routeConfig().value()->usesVhds())) {
+
+  if (callbacks_->route() != nullptr) {
```

Comment:
It is a valid configuration when the on_demand filter is enabled but no vhds section is present in RDS config: the check was meant to handle such a config. I couldn't find if this is being handled someplace else now.

Reply:
nm, found it here: https://github.com/envoyproxy/envoy/pull/12640/files#diff-d70ea2d9706e0246700148b2e2bb63ceR1571.

Reply:
Yes, everything is done inside active_stream now.

Reply:
One of the objectives of having the on-demand filter was to split out functionality from HCM to a standalone filter.

Reply:
some clarification here: on_demand vhds introduced the RouteConfigRequester as a member in the ActiveStream, which deals with "on-demand" config loading, I think it's fine to move the on-demand config loading logic to that object. this way requires less plumbing to expose the "route config provider" or the "scoped route provider" to be available at the filter. (I vote for removing routeConfig() from the filter interface)